src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
> The idea that dual numbers are an odd kind of complex number seems completely wrong to me. 

Dual numbers are a special case of the class of [**generalized** complex numbers](http://people.rit.edu/harkin/research/articles/generalized_complex_numbers.pdf) (as are the "normal" complex numbers, where i² = -1).

> The fact that you can, in some cases, use complex numbers to do automatic differentiation of analytic functions is just an accident and doesn't mean that dual numbers are a variation on complex numbers.

Agreed. I'm not sure whether you wrote this in response to me or not - but if you did, I'll mention that my idea involving `ForwardDiffComplex` was in regards to applying dual numbers for differentiating complex functions, not using complex numbers for AD.

> This is because (ℝ[ɛ]/ɛ²)/ɛ ≅ ℝ whereas (ℝ[i]/(i²+1))/i ≅ {0} – i.e. you can mod out the dual numbers by ɛ and get back the real number line, whereas if you mod out the complex numbers by i you've collapsed 0 and -1, which forces all of the real number line to a single point.

Interesting point